### PR TITLE
Share Network enum in both binaries

### PR DESCRIPTION
**Pull request by klochowicz**
Remove unnecessary duplication to ease maintenance

### CODE REVIEW

**Review by bonomat — APPROVED** [review]

File: shared-bin/src/network.rs [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    fn default() -> Self {
+        Network::Mainnet {
+            electrum: MAINNET_ELECTRUM.to_string(),
+            withdraw: None,
+        }
+    }
+}
+
+#[derive(Subcommand, Clone)]
+pub enum Withdraw {
```
Comment (luckysori) [inline shared-bin/src/network.rs]: 🔧 I think the withdraw subcommand doesn't belong in the `network` module. Perhaps we should create a `cli` module and put it there.

File: taker/src/main.rs [hunk 2]
```diff
@@ -427,7 +306,7 @@ async fn main() -> Result<()> {
         Err(_) => Environment::Binary,
     };
 
-    let price_feed_network = network.price_feed_network();
+    let bitmex_network = network.bitmex_network();
```
Comment (luckysori) [inline taker/src/main.rs:310]: 🙃 Not worth changing now, but it's easier to review this PR if renaming this method happens in a separate commit.

**Review by luckysori — COMMENTED** [review]
LGTM, but I would  move `Withdraw` to a different module.

File: taker/src/main.rs (hunk 2 above)
Comment (klochowicz) [reply taker/src/main.rs:310]: will keep in mind next time!

File: shared-bin/src/network.rs (hunk 1 above)
Comment (klochowicz) [reply shared-bin/src/network.rs]: just to be clear: would renaming `network` to `cli` be acceptable?

Comment (luckysori) [reply shared-bin/src/network.rs]: I think `Network` and `Withdraw` don't belong together.

Comment (klochowicz) [reply shared-bin/src/network.rs]: well, `Withdraw` is *part* of `Network` right now, isn't it? 🤔

I will move `Withdraw` to a separate `withdraw` module for now as a first step towards separating them.

Comment (luckysori) [reply shared-bin/src/network.rs]: Nvm, your proposal is better: renaming `network` to `cli`. I failed to notice that `Network` was actually a CLI subcommand. Sorry for the confusion!

**Pull request merged**